### PR TITLE
Adds automated functional testing

### PR DESCRIPTION
**Pull request by gcardonag**
# Pull Request

## Change Summary

- Adds local testing via Cypress
- Adds the ability to run automated functional tests using Cypress and GitHub actions
- Adds tests for most basic Desktop and Mobile scenarios

## Change Reason

This change is being made as a start to an effort to reduce Pull Request (PR) review times. Once all current functional test scenarios are covered by automated tests, PR reviewers can trust that site functionality works and focus their reviews on any net-new functionality and code structure and formatting.

## Verification [Optional]

No video available, but a sample GitHub action run with testing completed can be seen here: https://github.com/phlask/phlask-map/actions/runs/9572807129

Related Issue: N/A

### CODE REVIEW

File: cypress/e2e/desktop/crowdsourcing.cy.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      cy.get('input[name="drinkingFountain"]').click({force: true})
+      cy.get('input[name="sink"]').click()
+      cy.get('input[name="sodaMachine"]').click()
+      cy.get('input[name="waterCooler"]').click()
+      cy.get('svg[data-testid="ExpandMoreIcon"]').click({force: true})
+      cy.get('svg[data-testid="ArrowForwardIosIcon"]').click()
+      cy.get('input[name="handicapAccessible"]').click()
+      cy.get('input[name="waterVesselNeeded"]').click()
+      cy.get('textarea[name="guidelines"]').type("Cypress Test")
+      // cy.get('input[type="submit').click()
```
Comment (RNR1) [inline cypress/e2e/desktop/crowdsourcing.cy.js:41]: add TODO

File: src/components/Filter/Filter.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ const FilterTags = props => (
           props.handleTag(0, props.resourceType, props.index, key);
           props.forceUpdate();
         }}
+        data-cy={"filter-option-"+tag}
```
Comment (RNR1) [inline src/components/Filter/Filter.js]: let's use a template string

File: src/components/Filter/Filter.js [hunk 3]
```diff
@@ -47,6 +48,7 @@ const FilterTagsExclusive = props => (
           props.handleTag(1, props.resourceType, props.index, key);
           props.forceUpdate();
         }}
+        data-cy={"filter-option-"+tag}
```
Comment (RNR1) [inline src/components/Filter/Filter.js]: +

File: src/components/ReactGoogleMaps/ReactGoogleMaps.js [hunk 4]
```diff
@@ -327,6 +327,9 @@ export class ReactGoogleMaps extends Component {
                     icon={{
                       url: phlaskMarkerIconV2(resource.resource_type, 56, 56)
                     }}
+                    // This is used for marker targetting as we are unable to add custom properties with this library.
+                    // We should eventually replace this so that we can still enable the use of screen readers in the future.
+                    title={'data-cy-' + index}
```
Comment (RNR1) [inline src/components/ReactGoogleMaps/ReactGoogleMaps.js]: +

**Review by RNR1 — APPROVED** [review]
LGTM + some nits

File: cypress/e2e/desktop/crowdsourcing.cy.js (hunk 1 above)
Comment (gcardonag) [reply cypress/e2e/desktop/crowdsourcing.cy.js:41]: Done! ✍️

**Pull request merged**